src/components/Employees/index.tsx

```diff
@@ -0,0 +1,214 @@
+import styles from './Employees.module.css';
+import employees from '@components/Employees/employeesData';
+import EmployeeItem from '@components/EmployeeItem';
+import { IEmployeeItem } from '@components/EmployeeItem/types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearch, faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+import { useSearchParams } from 'react-router-dom';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+function Employees() {
+    const { t } = useTranslation();
+    const [searchParams, setSearchParams] = useSearchParams({});
+    const [searchActiveButtons, setSearchActiveButtons] = useState<string[]>(
+        []
+    );
+
+    function searchButtonExists(searchParam: string): boolean {
+        return searchActiveButtons.some(
+            (searchBtn) => searchBtn === searchParam
+        );
+    }
+
+    function employeeExists(
+        searchParam: IEmployeeItem,
+        filteredEmployees: IEmployeeItem[]
+    ): boolean {
+        return filteredEmployees.some((searchBtn) => searchBtn === searchParam);
+    }
+
+    function addFilter(target: string, keyParam: string) {
+        searchParams.append(keyParam, target);
+        setSearchParams(searchParams);
+        if (!searchButtonExists(target)) {
+            setSearchActiveButtons([...searchActiveButtons, target]);
+        }
+    }
+
+    function findFilters(filterArray: string[], target: string) {
+        const foundFilters: string[] = [];
+        filterArray.forEach((filter) => {
+            if (filter !== target) {
+                foundFilters.push(filter);
+            }
+        });
+        return foundFilters;
+    }
+
+    function applyFilters(filters: string[], target: string) {
+        searchParams.delete(target);
+        setSearchParams(searchParams);
+        filters.forEach((filter) => {
+            searchParams.append(target, filter);
+        });
+        setSearchParams(searchParams);
+    }
+
+    function removeFilter(target: string, keyParam: string) {
+        if (searchButtonExists(target)) {
+            const newSearchActiveButtons = searchActiveButtons.filter(function (
+                item
+            ) {
+                return item !== target;
+            });
+            setSearchActiveButtons(newSearchActiveButtons);
+            const search = searchParams.getAll(keyParam);
+            const foundFilters = findFilters(search, target);
+            applyFilters(foundFilters, keyParam);
+        }
```

Comment:
By returning early you can reduce nesting:
```suggestion
        if (!searchButtonExists(target)) {
             return;
        }
        const newSearchActiveButtons = searchActiveButtons.filter(function (
            item
        ) {
            return item !== target;
        });
        setSearchActiveButtons(newSearchActiveButtons);
        const search = searchParams.getAll(keyParam);
        const foundFilters = findFilters(search, target);
        applyFilters(foundFilters, keyParam);
```